lib/functions/settings.lua

```diff
@@ -1,49 +1,23 @@
-local assign = import("../assign")
+local RenderSettings = import("./settings/RenderSettings")
 
 local Settings = {}
-
-setmetatable(Settings, {
-	__tostring = function()
-		return "Settings"
-	end,
-})
-
-local prototype = {}
-
-function prototype:GetFFlag(name)
-	return self.settings[name] or false
-end
-
-local metatable = {}
-metatable.type = Settings
-
-function metatable:__index(key)
-	local internal = getmetatable(self).internal
-
-	if internal[key] ~= nil then
-		return internal[key]
-	end
-
-	if prototype[key] ~= nil then
-		return prototype[key]
-	end
-
-	error(string.format("%s is not a valid member of Settings", tostring(key)), 2)
-end
+Settings.__index = Settings
+Settings.type = "Settings"
 
 function Settings.new(settings)
-	local internalInstance = {
+	local instance = {
 		settings = settings or {},
+		Rendering = RenderSettings.new(),
 	}
 
-	local instance = newproxy(true)
-
-	assign(getmetatable(instance), metatable)
-	getmetatable(instance).internal = internalInstance
-
+	setmetatable(instance, Settings)
```

Comment:
I think we should find a middleground between the previous `userdata` boilerplate and dropping down completely into a table.

Notably, the definition of `typeof` depends on custom types being `userdata`.